### ISSUE TITLE
New Resource Interface

### ISSUE THREAD

**Issue by rovvp** [marmelab/react-admin#5317]
This is more than likely related to how resources work with react-admin and no doubt brought up in some other context however to explain in plain English and without my own code samples. Purely use case scenarios rather than trouble I'm having. 

The existing List / Datagrid and associated components appear limited in when building a dashboard type interface with multiple resources on a single component items such as search / pagination / filters etc appear geared towards a one-one component (url) / resource delegation. 

When attempt to place several <List> / <Datagrid> components on a single interface (Container Component) the react-admin Resource / List analogy appears to have flaws. I understand this is by design and perhaps I'm missing a key part of the documentation however I feel a new type of component allowing for this may be justified? 

Hope I made my point clear and apologies if I am miss understanding  / missing documented examples where features are not about to be depreciated. 

If anyone can shed some light and call my ignorance out - feel free otherwise please consider the request.

**Comment by otaviobps** [comment]
The main problem I found when using react-admin was passing the props to the components. They use a behavior of cloning the props to the children, but when you use a different component that doesn't have this behavior in between their components, you can't use their components, like `<TextField>`. 

I had a hard time reading the documentation and couldn't find the solution for hours, until I found a hook that makes all the process very simple.

This is my code to a `Show` component:

```
export default function UserShow(props) {
    const { basePath, record, resource, version, loaded } = useShowController(props);

    const { t } = useTranslation('admin');

    const adminProps = {
        basePath,
        record,
        resource,
        version,
    };

    if (!loaded) {
        return <LoaderMaterial />;
    }

    return (
        <DashboardContainer>
            <Grid container spacing={4}>
                <Grid item xs={12}>
                    <InfoCard icon={<AccountCircleIcon />} title={t('User information')}>
                        <Grid item sm={6}>
                            <Typography gutterBottom>{t('Full name')}</Typography>
                            <FunctionField
                                {...adminProps}
                                render={(record) => `${record.firstName} ${record.lastName}`}
                            />
                        </Grid>
                        <Grid item sm={6}>
                            <Typography gutterBottom>{t('Email')}</Typography>
                            <TextField source="email" {...adminProps} />
                        </Grid>
                        <Grid item sm={6}>
                            <Typography gutterBottom>{t('Document type')}</Typography>
                            <TextField source="identity.idType" {...adminProps} />
                        </Grid>
                        <Grid item sm={6}>
                            <Typography gutterBottom>{t('Document number')}</Typography>
                            <TextField source="identity.number" {...adminProps} />
                        </Grid>
                        <Grid item sm={6}>
                            <Typography gutterBottom>{t('Document expiration')}</Typography>
                            <DateField source="identity.expiration" {...adminProps} />
                        </Grid>
                        <Grid item sm={6}>
                            <Typography gutterBottom>{t('Language')}</Typography>
                            <TextField source="language" {...adminProps} />
                        </Grid>
                        <Grid item sm={6}>
                            <Typography gutterBottom>{t('Referral code')}</Typography>
                            <TextField source="referral" {...adminProps} />
                        </Grid>
                    </InfoCard>
                </Grid>
                <Grid item xs={12}>
                    <InfoCard icon={<StoreIcon />} title={t('User orders')}>
                        <Grid item>
                            <ReferenceArrayField
                                source="orders"
                                reference="orders"
                                sortBy="createdAt"
                                sortByOrder="DESC"
                                link={false}
                                {...adminProps}
                            >
                                <SingleFieldList linkType="show">
                                    <ChipField source="shortId" color="primary" />
                                </SingleFieldList>
                            </ReferenceArrayField>
                        </Grid>
                    </InfoCard>
                </Grid>
                <Grid item xs={12}>
                    <InfoCard icon={<GroupIcon />} title={t('User recipients')}>
                        <Grid item>
                            <ReferenceArrayField
                                source="recipients"
                                reference="recipients"
                                sortBy="name"
                                sortByOrder="ASC"
                                link={false}
                                {...adminProps}
                            >
                                <Datagrid>
                                    <TextField source="name" />
                                    <TextField source="country" />
                                    <TextField source="documentNumber" />
                                    <TextField source="account" />
                                </Datagrid>
                            </ReferenceArrayField>
                        </Grid>
                    </InfoCard>
                </Grid>
            </Grid>
        </DashboardContainer>
    );
}
```

The `useShowController` hook have all necessary props needed by the components. I just added them to an object to be easier to pass to the components.

I hope this helps you and everybody else who has problems customizing the pages.

**Comment by fzaninotto** [comment]
I hope I understand your question well.

The main "page" controllers that you pass to a `Resource` (`List`, `Edit`, `Show`) are designed to display one main UI component per page. They grab the parameters for that component from the URL. 

But these "page" controllers are quite small. They rely on react-admin hooks that do most of the job of data fetching, grabbing data from cache, and dispatching side effects. 

That means you can "compose" several core UI components (like a list, a show view, etc) by using these hooks yourself, and passing the props directly to the components (not `List` or `Edit`, but rather `Datagrid` or `SimpleForm`).

See how we used that logic to build a dashboard in [the react-admin demo](https://marmelab.com/react-admin-demo/#/) ([source](https://github.com/marmelab/react-admin/tree/master/examples/demo/src/dashboard)):

![image](https://user-images.githubusercontent.com/99944/94648132-2588de00-02f2-11eb-96f1-8a33def91f63.png)

This is somehow explained in the [Querying the API](https://marmelab.com/react-admin/Actions.html) documentation chapter.

**Comment by rovvp** [comment]
Thanks for the responses guys. Both good points.

I think in my case I was trying to use multiple lists / data grids on a single component . I may just need to simplify the UI to look more like the dashboard example.

**Comment by otaviobps** [comment]
You can use how many components you want, in the order you want. Just make sure to pass the props to them (see my example) from the `useShowController` hook
